openlibrary/catalog/add_book/__init__.py

```diff
@@ -993,6 +1001,13 @@ def load(rec, account_key=None, from_marc_record: bool = False):
 
     normalize_import_record(rec)
 
+    # For recs with a non-ISBN ASIN, supplement the record with BookWorm data.
```

Comment:
I don't really understand what BookWorm data is. The code is appears to be looking for pre-exisiting "staged or pending" import items by ASIN regardless of the original source?

Reply:
I will fix that comment because it's actually supplementing with data from the `import_item` table, which could be ISBNdb data, it could be from Amazon's Product API, or another source. Thanks for pointing this out.

And indeed, for any `rec` with a `B*` ASIN, this will try to supplement that `rec` by using a match from the `import_item` table. This could easily be more limited if desired.

As for what BookWorm is, that's the name for the affiliate-server.